src/lib/Messages.js

```diff
@@ -255,67 +255,67 @@ class Messages {
     }
   };
 
-  tryfromBinary = <TType>(buffer: Buffer, typeName: string): ?TType => {
-      var result = null;
+  tryFromBinary = (buffer: Buffer, typeName: string): ?mixed => {
+      let result = null;
       try {
         result = this.fromBinary(buffer, typeName);
-      } catch (exception) {
-        logger.error('Could not parse type: ${typeName} ${buffer}', exception);
+      } catch (error) {
+        logger.error('Could not parse type: ${typeName} ${buffer}', error);
       }
       return result;
   };
 
-  fromBinary = <TType>(buffer: Buffer, typeName: string): TType => {
```

Comment:
In the last two cases I would just prefer to explicitly cast to `Buffer` or whatever so that flow passes.

Also, there are two functions you could use on `BufferReader` that may help you pass the flow tests -- `readBuffer` and `readString`